File: Integration Examples/eAPI v1/java/eAPI-v1-connector/src/main/java/cz/monetplus/mips/eapi/v1/connector/entity/SignBase.java
package cz.monetplus.mips.eapi.v1.connector.entity;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

import org.joda.time.DateTime;

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
public abstract class SignBase extends ApiBase implements Signable {

	private static final long serialVersionUID = -3825192932302805075L;
	
	@XmlElement(nillable=false)
	public String dttm;

	@XmlElement(nillable=false)
	public String signature;

	public void fillDttm() {
		this.dttm = DateTime.now().toString(DTTM_FORMAT);
	}

}

File: Integration Examples/eAPI v1.6/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/SignBase.cs
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects
{
    public abstract class SignBase : ApiBase
    {
        [JsonProperty("dttm")]
        public string DateTime { get; set; }
        
        [JsonProperty("signature")]
        public string Signature { get; set; }

        public abstract string ToSign();
    }
}
